tools/gfx/debug-layer.h

```diff
@@ -317,7 +317,8 @@ class DebugResourceCommandEncoder : public UnownedDebugObject<IResourceCommandEn
     virtual SLANG_NO_THROW void SLANG_MCALL
         uploadBufferData(IBufferResource* dst, size_t offset, size_t size, void* data) override;
     virtual SLANG_NO_THROW void SLANG_MCALL writeTimestamp(IQueryPool* pool, SlangInt index) override;
-
+    virtual SLANG_NO_THROW void SLANG_MCALL textureBarrier(ITextureResource* texture, ResourceState src, ResourceState dst) override;
```

Comment:
I think we should consider making the interface similar to D3D12, where we have a `struct ResourceBarrier`, and a `resourceBarrier(uint32_t count, ResourceBarrier* barriers)` function to reduce the amount of function calls needed for a large number of barriers.

Reply:
Another concern is whether or not we want a resource to be in a state that is a combination of `ResourceState` values, and that will certainly complicate the translation logic.

Reply:
Yes, I think that interface would be better in this case because it's a closer mapping to VK/DX. I can make that change.

Reply:
Let's change the interface then. We can do it in the next PR.